File: cpp/IceGrid/sessionActivation/HelloI.cpp
// Copyright (c) ZeroC, Inc.

#include "HelloI.h"
#include <Ice/Ice.h>
#include <iostream>

using namespace std;

HelloI::HelloI(string_view name) : _name(name) {}

void
HelloI::sayHello(const Ice::Current&)
{
    cout << _name << " says Hello World!" << endl;
}

File: csharp/IceGrid/simple/HelloI.cs
// Copyright (c) ZeroC, Inc.

using Demo;

public class HelloI(string name) : HelloDisp_
{
    public override void sayHello(Ice.Current current)
    {
        Console.WriteLine(_name + " says Hello World!");
    }

    public override void shutdown(Ice.Current current)
    {
        Console.WriteLine(_name + " shutting down...");
        current.adapter.getCommunicator().shutdown();
    }

    private string _name = name;
}
